File: backend/src/main/java/com/back/domain/care/controller/CareController.java
package com.back.domain.care.controller;

import com.back.domain.care.dto.request.CareRequestDto;
import com.back.domain.care.dto.response.CareResponseDto;
import com.back.domain.care.service.CareService;
import com.back.global.common.ApiResponse;
import io.swagger.v3.oas.annotations.Operation;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api")
@RequiredArgsConstructor
public class CareController {

    private final CareService careService;

    @PostMapping("/applies/care")
    @Operation(summary = "돌봄 신청", description = "돌봄 신청을 처리합니다.")
    public ResponseEntity<ApiResponse<CareResponseDto>> applyCare(
            @RequestBody CareRequestDto careRequestDto) {
        CareResponseDto careResponseDto = careService.applyCare(careRequestDto);
        // 알람은 추후 구현
        return ResponseEntity.status(HttpStatus.OK).body(
                ApiResponse.success(careResponseDto)
        );
    }
}

File: backend/src/main/java/com/back/domain/care/controller/CareController.kt
package com.back.domain.care.controller

import CareRequestDto
import com.back.domain.care.dto.response.CareResponseDto
import com.back.domain.care.service.CareService
import com.back.global.common.ApiResponse
import io.swagger.v3.oas.annotations.Operation
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class CareController(
    private val careService: CareService
) {

    @PostMapping("/applies/care")
    @Operation(summary = "돌봄 신청", description = "돌봄 신청을 처리합니다.")
    fun applyCare(
        @RequestBody careRequestDto: @Valid CareRequestDto,
        @AuthenticationPrincipal userDetails: UserDetails
    ): ResponseEntity<ApiResponse<CareResponseDto>> {
        val careResponseDto = careService.applyCare(careRequestDto, userDetails.username)
        return ResponseEntity.ok(ApiResponse.success(careResponseDto))
    }
}
